tools/tcpretrans.py

```diff
@@ -70,6 +74,24 @@
 };
 BPF_PERF_OUTPUT(ipv6_events);
 
+// separate flow keys per address family
+struct ipv4_fl_k_t {
+    // XXX: switch some to u32's when supported
+    u64 saddr;
+    u64 daddr;
+    u64 lport;
+    u64 dport;
+};
+BPF_HASH(ipv4_count, struct ipv4_fl_k_t);
```

Comment:
How about "u32 saddr; u32 daddr; u16 lport; u16 dport;"? There are no holes here and it should work?

Reply:
Took that from 

struct ipv4_data_t {
    // XXX: switch some to u32's when supported
    u64 pid;
    u64 ip;
    u64 saddr;
    u64 daddr;
[...]

which was already present. In principle, though ...